lnst/Recipes/ENRT/GeneveOvsNetnsTunnelRecipe.py

```diff
@@ -0,0 +1,476 @@
+
+                    for src_addr, dst_addr in zip(endpoint1_ips, endpoint2_ips):
+                        pconf = PingConf(client = endpoint1.netns,
+                                         client_bind = src_addr,
+                                         destination = endpoint2.netns,
+                                         destination_address = dst_addr,
+                                         count = self.params.ping_count,
+                                         interval = self.params.ping_interval,
+                                         size = self.params.ping_psize,
+                                         )
+
+                        ping_evaluators = self.generate_ping_evaluators(
+                                pconf, endpoints)
+
+                        pconf.register_evaluators(ping_evaluators)
+                        ping_conf_list.append(pconf)
+
+                        if self.params.ping_bidirect:
+                            ping_conf_list.append(self._create_reverse_ping(pconf))
+
+                if ping_conf_list:
+                    yield ping_conf_list
+
+    def generate_ping_endpoints(self, config):
+        """
+        The ping endpoints for this recipe are the IP/IPv6 addresses configured
+        on the veth_peer devices in the network namespaces
+        """
+        return [
+            PingEndpoints(first, second)
+            for first, second
+            in zip(
+                [self.matched.host1.__getattr__(f"netns{flow_id}").veth_peer for flow_id in range(self.params.flow_count)],
+                [self.matched.host2.__getattr__(f"netns{flow_id}").veth_peer for flow_id in range(self.params.flow_count)]
+            )
+        ]
+
+    def generate_perf_endpoints(self, config: EnrtConfiguration) -> list[Collection[EndpointPair[IPEndpoint]]]:
+        """
+        The perf endpoints for this recipe are the devices in individual network
+        namespaces that are configured with IP addresses of the tunnelled networks.
+        """
+        """
+        return [
+            [
+                EndpointPair(first, second)
+                for first, second
+                in zip(
+                    [self.matched.host1.__getattr__(f"netns{flow_id}").veth_peer for flow_id in range(self.params.flow_count)],
+                    [self.matched.host2.__getattr__(f"netns{flow_id}").veth_peer for flow_id in range(self.params.flow_count)]
+                )
+            ]
+        ]
+        """
+        return [
+            ip_endpoint_pairs(
+                config,
+                *(
+                    (first, second)
+                    for (first, second)
+                    in zip(
+                        [self.matched.host1.__getattr__(f"netns{flow_id}").veth_peer for flow_id in range(self.params.flow_count)],
+                        [self.matched.host2.__getattr__(f"netns{flow_id}").veth_peer for flow_id in range(self.params.flow_count)]
+                    )
```

Comment:
i feel like there should be a helper "tunnel getter" method somewhere... either a generator or these can maybe be created during the tunnel creation and stored in the test wide configuration `config` object.

this would probably make the code easier to understand and maintain...

Reply:
Fixed by using `config.tunneled_endpoints`